great_expectations_cloud/agent/actions/generate_data_quality_check_expectations_action.py

```diff
@@ -150,12 +154,12 @@ def _retrieve_asset_from_asset_name(
             # TODO - see if this can be made more specific
             raise RuntimeError(f"Failed to retrieve asset: {e}") from e  # noqa: TRY003 # want to keep this informative for now
 
-        return data_asset
+        return data_asset  # type: ignore[no-any-return]  # unable to narrow types strictly based on names
 
-    def _get_metrics(self, data_asset: DataAsset) -> tuple[MetricRun, UUID]:
+    def _get_metrics(self, data_asset: DataAsset[Any, Any]) -> tuple[MetricRun, UUID]:
         batch_request = data_asset.build_batch_request()
         metric_run = self._batch_inspector.compute_metric_list_run(
-            data_asset_id=data_asset.id,
+            data_asset_id=data_asset.id or uuid4(),  # id can be None
```

Comment:
Is it every None in practice? That is, we didn't do this before; did we use to crash? I'm wondering if it's better if we check and raise if it's None.

Reply:
I'm not sure. Maybe in this case it's better to raise.